keps/sig-architecture/3784-database-of-names/README.md

```diff
@@ -0,0 +1,449 @@
+type RecordList []Record
+
+// Record describes a single reserved name.
+type Record struct {
+    // Key is the full name being claimed, including the prefix.
+    // Required
+    Key string
+
+    // SIG specifies which SIG owns this name.  Multiple SIGs might use a name,
+    // but only one can own it.
+    // Required
+    SIG SIG
+
+    // Description provides human-friendy information about what this name
+    // means and how/when to use it.
+    // Required
+    Description string
+
+    // DocsLink provides an HTTP link to more information about this name.
+    // Optional
+    DocsLink string
+
+    // Context specifies where this name might be used.  Names which can be
+    // used in more than one place should define multiple name records.
+    // Subsequent fields' requiredness depend on the context.
+    Context Context
+
+    // ValueType specifies what "inner" type(s) of value are expected for
+    // contexts which require a value.  For example, annotation values are
+    // always strings, but the content of those strings might be expected to be
+    // integers, booleans, or even JSON-encoded structs.  Some names accept
+    // multiple types.
+    // Required if context demands
+    ValueType []ValueType
+
+    // Resources specifies which API resource(s) this name might be used with,
+    // in "apigroup/resource" syntax.
+    // Required if context demands
+    Resources []string
+}
+
+// SIG is a Kubernetes special interest group name.
+type SIG string
+
+const (
+    SIGNetwork = SIG("Network")
+    SIGNode = SIG("Node")
+    SIGApps = SIG("Apps")
+    SIGStorage = SIG("Storage")
+    SIGArchitecture = SIG("Architecture")
+    // more...
+)
+
+// Context describes where a name might be used.
+type Context string
+
+const (
+    ContextLabel = Context("Label")             // key+value
+    ContextAnnotation = Context("Annotation")   // key+value
+    ContextTaint = Context("Taint")             // key
+    ContextCondition = Context("Condition")     // key
+    ContextAppProtocol = Context("AppProtocol") // key
+    ContextAppProtocol = Context("ApiGroup")    // key
+    // more...
+)
+
+// ValueType describes the datatype of a value for a key name.
+type ValueType string
+
+const (
+    ValueTypeInt = ValueType("Int")
+    ValueTypeString = ValueType("String")
+    ValueTypeBool = ValueType("Bool")
+    ValueTypeJSONObject = ValueType("JSONObject")
+    ValueTypeJSONList = ValueType("JSONList")
+    // more...
+)
+```
+
+We can publish this schema as Go code or something else if there is a better or
+more standard notation.  We can validate all names against this schema.
+
+<<[UNRESOLVED enumerate resources ]>>
+Should we enumerate resources?  Good for validation, but a bit tedious.
+<<[/UNRESOLVED]>>
+
+#### Why YAML
+
+YAML is the lingua franca of Kubernetes.  We propose to use a subset of YAML
+which is closer to JSON and easier to avoid whitespace errors, while still
+allowing comments.
```

Comment:
Why comments? Shouldn't every piece of relevant information be stored in a field?

Reply:
Good insight, but doesn't change my preference that comments be POSSIBLE :)